RealmSwift/Realm.swift

```diff
@@ -1240,6 +1240,18 @@ public final class Realm {
         }
     }
 
+    public func async<C: RangeReplaceableCollectionType where C.Generator.Element: Object>(
```

Comment:
@jpsim I'm actually not sure how much sense it makes to offer a variant supporting polymorphic containers. In what situations would a user want to pass a non-array collection of Realm objects across thread boundaries?

- `Set`: It's not sensible to keep a set of objects since they're mutable so their hash value is subject to change. I guess a user might treat their objects as if they're immutable, or they must base the hash value solely on the object identifier, but these seem like less common situations.
- `Dictionary`: Though I do think it's reasonable to pass dictionaries across thread boundaries, this would _not_ be supported by the above overload since `Dictionary` isn't a `RangeReplaceableCollectionType`. I'm not sure how worthwhile such an overload is though. It would have to be clear that _only_ Realm objects are handed over in a thread safe way. And I think there are better alternatives anyway.
- `Sequence`: We can't pass an arbitrary sequence across thread boundaries without exposing it as either an `Array` or an `AnySequence` on the other end. It seems like the user should just build an array before passing, otherwise we may mislead the user into thinking that the sequenced is not traversed on handover (or that it is traversed lazily on access).

What are your thoughts?